File: src/server/SurfaceRect.cs
public class SurfaceRect
{
	public uint left;
	public uint top;
	public uint right;
	public uint bottom;
	public byte[][] data;

	public SurfaceRect (uint left, uint top, uint right, uint bottom, byte[][] data)
	{
		this.left = left;
		this.top = top;
		this.right = right;
		this.bottom = bottom;
		this.data = data;
	}

	public byte GetSurfaceAt (uint x, uint y)
	{
		if (x >= left && x <= right && y >= top && y <= bottom)
		{
			return this.data[x-left][y-top];
		}
		else
		{
			return 255;
		}
	}

}

File: src/server/SurfaceRect.php
<?php
class SurfaceRect {
	public $entity = "surface";
	public $left;
	public $top;
	public $right;
	public $bottom;
	public $data;

	public function __construct ($left, $top, $right, $bottom, $data) {
		$this-> left = $left;
		$this-> top = $top;
		$this-> right = $right;
		$this-> bottom = $bottom;
		$this-> data = $data;
	}
}
